config/routes.rb

```diff
@@ -4,3 +4,4 @@
   get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
   get '/user/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
 end
+ 
```

Comment:
- [ ] [optional] @tomdan-ai Great work implementing all changes requested, however, I would recommend using resources for your routes, this will make it easier to read your code and to expand scale the app in future.
```suggestion
 root 'users#index'

  resources :users, only: [ :show, :index ] do
    resources :posts, only: [ :show, :index ] do 
    end
  end
```